### ISSUE TITLE
Concat with an Observable which has multiple subscribers make the Observable run more than one time

### ISSUE THREAD

**Issue by luxl** [Cysharp/R3#244]
I know it might be hard to understand the title, let's see the code:
```C#
var doScan = Observable.FromAsync(async (token) =>
                                  {
                                      Console.WriteLine("scan start");
                                      await Task.Delay(TimeSpan.FromSeconds(3));
                                      Console.WriteLine("scan end");
                                      return 5;
                                  });

var doCalc = Observable.FromAsync(async (token) =>
                                  {
                                      Console.WriteLine("calc start");
                                      await Task.Delay(TimeSpan.FromSeconds(3), token);
                                      Console.WriteLine("calc end");
                                      return 10;
                                  });

var countDown = Observable.Interval(TimeSpan.FromMilliseconds(300)).Index().Select(v => v > 9 ? 9 : v);

var work = doScan.Select(_ => doCalc).Switch().Replay(1).RefCount();
countDown.TakeUntil(work.LastAsync()).Concat(work.TakeLast(1)).Subscribe(v => Console.WriteLine($"progress: {v}"));
```
the above is basicly a progress, from 0 to 10, it will stay at 9 until the work is done.
here is it's output:
```bash
scan start
progress: 0
progress: 1
progress: 2
progress: 3
progress: 4
progress: 5
progress: 6
progress: 7
progress: 8
scan end
calc start
progress: 9
progress: 9
progress: 9
progress: 9
progress: 9
progress: 9
progress: 9
progress: 9
progress: 9
progress: 9
calc end
progress: 10
scan start
scan end
```
As we can see, when we reach `progress: 10`, it print `scan start` again, which should not happen. If I change the last line code to:
```C#
// replace Concat with Merge
countDown.TakeUntil(work.LastAsync()).Merge(work.TakeLast(1)).Subscribe(v => Console.WriteLine($"progress: {v}"));
```
it will work as expected

================updata=============
If I add `Publish().RefCount()` to `doScan`, and still use `Concat`, it will work as expected. But I don't need to do so in RX.NET, so what's wrong? I'm a little lost.

**Comment by neuecc** [comment]
If you change `TakeUntil(work.LastAsync())` to `TakeUntil(work.TakeLast(1))`, will it work?

It seems that the timing of RefCount is different for Tasks and Observables.
In the case of Tasks (LastAsync), the Subscribe of TakeLast(1) is called after the RefCount becomes 0 when Dispose is executed at the completion of LastAsync, so the source is resubscribed.
In the case of Observable (TakeLast(1)), the Subscribe was called before Dispose, so there was no re-subscription of the source.

I think that in the case of Task, some of the work is being sent to the ThreadPool, which is probably why there is a difference in timing.
However, in this case, the desired behavior is the same as for Observable, so I would like to investigate this a little further.
Thank you.

**Comment by luxl** [comment]
Yes, it worked by using `TaksLast(1)`

**Comment by neuecc** [comment]
Due to a bug in RefCount, the internal count would temporarily become incorrect when the target was in the OnCompleted state.
This has been fixed in v1.2.6.
